client/beefy/rpc/src/lib.rs

```diff
@@ -20,95 +20,78 @@
 
 #![warn(missing_docs)]
 
-use std::sync::Arc;
-
-use sp_runtime::traits::Block as BlockT;
-
-use futures::{FutureExt, SinkExt, StreamExt};
-use jsonrpc_derive::rpc;
-use jsonrpc_pubsub::{manager::SubscriptionManager, typed::Subscriber, SubscriptionId};
-use log::warn;
-
 use beefy_gadget::notification::BeefySignedCommitmentStream;
+use futures::{future, task::Spawn, FutureExt, StreamExt};
+use jsonrpsee::{
+	proc_macros::rpc,
+	types::{Error as JsonRpseeError, RpcResult},
+	SubscriptionSink,
+};
+use log::warn;
+use sc_rpc::SubscriptionTaskExecutor;
+use sp_runtime::traits::Block as BlockT;
 
 mod notification;
 
 /// Provides RPC methods for interacting with BEEFY.
-#[rpc]
+#[rpc(client, server, namespace = "beefy")]
 pub trait BeefyApi<Notification, Hash> {
-	/// RPC Metadata
-	type Metadata;
-
 	/// Returns the block most recently finalized by BEEFY, alongside side its justification.
-	#[pubsub(
-		subscription = "beefy_justifications",
-		subscribe,
-		name = "beefy_subscribeJustifications"
+	#[subscription(
+		name = "subscribeJustifications",
+		aliases = ["beefy_justifications"],
+		item = Notification,
 	)]
-	fn subscribe_justifications(
-		&self,
-		metadata: Self::Metadata,
-		subscriber: Subscriber<Notification>,
-	);
-
-	/// Unsubscribe from receiving notifications about recently finalized blocks.
-	#[pubsub(
-		subscription = "beefy_justifications",
-		unsubscribe,
-		name = "beefy_unsubscribeJustifications"
-	)]
-	fn unsubscribe_justifications(
```

Comment:
DQ: Why is this removed?

Reply:
the subscription attribute generates `an unsubscribe method` for us so no need to have a separate `unsubscription` in the proc macros.

For example

```rust
// namespace `beefy`
#[subscription(
		name = "subscribeJustifications",
		aliases = ["beefy_justifications"],
		item = Notification,
	)]
	fn subscribe_justifications(&self) -> RpcResult<()>;
```

it generates the methods `beefy_subscribeJustifications` and `beefy_unsubscribeJustification` so if you need a different name for the `unsubscription method` it possible via `unsubscribe_aliases = ["other_name"]`